axlearn/common/attention.py

```diff
@@ -1216,18 +1216,37 @@ class Config(BaseLayer.Config):
         dim: Required[int] = REQUIRED  # The dimensionality of the positional embedding.
         theta: float = 10000.0  # The scale of base frequency.
 
-    def forward(self, positions: Tensor) -> Tensor:
+    def default_query_positions(self, max_seq_len: int) -> Tensor:
```

Comment:
```suggestion
    def _default_query_positions(self, max_seq_len: int) -> Tensor:
```
Users should pass `max_seq_len` rather than calling this method publicly.

Reply:
There might be situations that we want to access the default query positions from the outside, such as getting the default positions and computing the positions based on it before passing it to `forward`. Therefore we want to keep this public.

Reply:
Do we expect subclasses to override this method?

If not, it will be more readable for callers to call `jnp.arange` directly given the implementation is only one line.

Reply:
Yes we do expect subclasses to override this method. One example is that we might want to specify the rotation start index and rotation end index for this embedding class. And the default embedding positions will be different then.

Reply:
Thanks for the explanation. Given this, it's reasonable to consolidate the position computation logic in this class.